lightning/backend/backend.go

```diff
@@ -212,6 +246,61 @@ func (be Backend) FetchRemoteTableModels(ctx context.Context, schemaName string)
 	return be.abstract.FetchRemoteTableModels(ctx, schemaName)
 }
 
+func (be Backend) FlushAll() error {
+	return be.abstract.FlushAllEngines()
+}
+
+// CheckDiskQuota verifies if the total engine file size is below the given
+// quota. If the quota is exceeded, this method returns an array of engines,
+// which after importing can decrease the total size below quota.
+func (be Backend) CheckDiskQuota(quota int64) (
+	largeEngines []uuid.UUID,
+	inProgressLargeEngines int,
+	totalDiskSize int64,
+	totalMemSize int64,
+) {
+	sizes := be.abstract.EngineFileSizes()
+	sort.Slice(sizes, func(i, j int) bool {
+		a, b := &sizes[i], &sizes[j]
+		if a.IsImporting != b.IsImporting {
+			return a.IsImporting
```

Comment:
What does it mean? Do you want to sort all importing engine  before other engines?

Reply:
Yes. We can't flush engines that are already importing, but their size do count towards the disk quota.